x-pack/plugins/security/server/routes/api/public/privileges/get.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { PrivilegeMap } from 'x-pack/plugins/security/common/model';
+
+export function initGetPrivilegesApi(
```

Comment:
My only reason to make this public is to enable API consumers to get a list of available privileges. Otherwise our role management API is going to be rather difficult to use without a bit of reverse-engineering on their side. I do agree that they don't need to be concerned with the "actions" though.

Do you think we should have a separate public API to enumerate just the privileges? The only downside there is that we wouldn't be using it at all, it would only be for other API consumers.

Reply:
> Do you think we should have a separate public API to enumerate just the privileges? The only downside there is that we wouldn't be using it at all, it would only be for other API consumers.

The only other idea that I have is to use a flag to return the action information, which we could take advantage of internally, but not document for "public consumption".

Reply:
I'll do that!